src/main/java/com/gmail/inverseconduit/chat/StackExchangeChat.java

```diff
@@ -52,19 +74,96 @@ public StackExchangeChat() {
      */
     @Override
     public boolean login(ProviderDescriptor descriptor, CredentialsProvider credentials) {
-        HtmlPage loginPage = getLoginPage(descriptor);
-        if (null == loginPage) { return false; }
+        if (descriptor.getDescription().toString().matches("(?!.*?(?:stackoverflow|meta)).+"))
```

Comment:
Why not make stackexchange the if-case and handle the problematic login in a separate method?